### PR TITLE
KT-20357: Add sample for Collections distinct and distinctBy methods

### PR DESCRIPTION
**Pull request by TillmannBerg**

### CODE REVIEW

File: libraries/tools/kotlin-stdlib-gen/src/templates/Sets.kt [hunk 1]
```diff
@@ -64,6 +64,7 @@ object SetOps : TemplateGroupBase() {
             returns("Sequence<T>")
             body { "return this.distinctBy { it }" }
         }
+        sample("samples.collections.Collections.Sets.distinctAndDistinctBy")
```
Comment (ilya-g) [inline libraries/tools/kotlin-stdlib-gen/src/templates/Sets.kt]: It'd be better to place the sample into `Collections.Transformations` class: `distinct` is a general collection operation, the fact that it uses a set to filter out duplicates is an implementation detail.

Comment (TillmannBerg) [reply libraries/tools/kotlin-stdlib-gen/src/templates/Sets.kt]: Are you sure? The function definitions are in a `SetOps` object in a `Sets.kt` file. To me the `Sets` class would be the place I'd expect the sample. But if you could confirm that you would prefer the sample in the `Transformations` section, I will move it.

Comment (ilya-g) [reply libraries/tools/kotlin-stdlib-gen/src/templates/Sets.kt]: Yes. The result functions generated from this template are placed in `_Arrays.kt`, `_Collections.kt`, `_Sequences.kt` and so on, therefore `distinct(By)` is a generic collection operation.

Comment (TillmannBerg) [reply libraries/tools/kotlin-stdlib-gen/src/templates/Sets.kt]: Done, I moved the sample to `collections.Transformations`.

**Pull request merged**